src/lib.rs

```diff
@@ -556,33 +602,51 @@ macro_rules! __impl_bitflags {
 
             /// Convert from underlying bit representation, dropping any bits
             /// that do not correspond to flags.
-            #[inline]
-            pub fn from_bits_truncate(bits: $T) -> $BitFlags {
-                $BitFlags { bits } & $BitFlags::all()
+            __fn_bitflags! {
+                #[inline]
+                pub fn from_bits_truncate(bits: $T) -> $BitFlags {
+                    $BitFlags { bits: bits & $BitFlags::all().bits }
+                }
             }
 
             /// Returns `true` if no flags are currently stored.
-            #[inline]
-            pub fn is_empty(&self) -> bool {
-                *self == $BitFlags::empty()
+            __fn_bitflags! {
+                #[inline]
+                pub fn is_empty(&self) -> bool {
+                    self.bits() == $BitFlags::empty().bits()
+                }
             }
 
             /// Returns `true` if all flags are currently set.
-            #[inline]
-            pub fn is_all(&self) -> bool {
-                *self == $BitFlags::all()
+            __fn_bitflags! {
+                #[inline]
+                pub fn is_all(&self) -> bool {
+                    self.bits == $BitFlags::all().bits
+                }
             }
 
             /// Returns `true` if there are flags common to both `self` and `other`.
-            #[inline]
-            pub fn intersects(&self, other: $BitFlags) -> bool {
-                !(*self & other).is_empty()
+            __fn_bitflags! {
+                #[inline]
+                pub fn intersects(&self, other: $BitFlags) -> bool {
+                    !$BitFlags{ bits: self.bits & other.bits}.is_empty()
+                }
             }
 
             /// Returns `true` all of the flags in `other` are contained within `self`.
-            #[inline]
-            pub fn contains(&self, other: $BitFlags) -> bool {
-                (*self & other) == other
+            __fn_bitflags! {
+                #[inline]
+                pub fn contains(&self, other: $BitFlags) -> bool {
+                    (self.bits & other.bits) == other.bits
+                }
+            }
+
+            /// Joins to set of flags into a new one
+            __fn_bitflags! {
```

Comment:
Shall we pull this new function out for now so we can review just the new machinery for `const` functions?

Reply:
I've removed this function by now. I can send it in a different PR though, without it the const functionality is not very useful since you can only check 1 flag at a time